docs/design/value-attribution.md

```diff
@@ -0,0 +1,79 @@
+# Value Attribution
+
+### Brandon Iglesias, Dennis Coyle
+
+
+## Definitions
+
+*Attribution* - _Acknowledgment for establishing the relationship between Storj and a customer_
+
+## OVERVIEW
+
+When our partners bring data to the Tardigrade network we want the ability to attach attribution for the data. 
+
+Our partners will have connectors that their customers will use to store data on the Tardigrade network. When one of their customers uses the connector to store data in a bucket on the Tardigrade network, we want to give that partner attribution for the data that is stored in it.
+
+## GOALS
+
+
+* A generic library that can be used in all  Golang third party connectors
+
+* Data attribution on a per bucket basis
+
+* Ability to calculate how much data was stored in the bucket for a given amount of time. 
+
+* Ability to calculate attributed data bandwidth on the Storj Network
+
+
+## NON GOALS
+
+* Attributions per object
+
+* Multiple attributions per bucket
+
+* Attributions per project
+
+* Attributions per satellite account
+
+
+
+## SCENARIOS
+
+
+* A client points partner connector A to a bucket with NO data in it.
+
+	* The partner receives the attribution for all of the data that the client stores and retrieves from it. 
+
+	* The bucket must be empty when the partner connector is connected to it. 
+
+* A client points partner connector A to an existing bucket that NO OTHER connector has pointed to but has data in it. 
+
+	* The partner does NOT receive attribution for data stored or retrieved from that bucket. 
+
+* A client points partner connector A to an existing bucket that partner connector B had already been pointed to. 
+
+	* Partner connector B receives the attribution for all of the data that is stored and retrieved from the bucket. 
+
+## DESIGN
+### Connector
+Each partner will have a registered uuid that will identify a partners connector on the Storj network. Each connector will implement the S3 client interface to allow operations through the connector.  When a user uploads data to a specified bucket through the connector,  the connector will include the partner uuid in the context of the request. Before an upload occurs, the uplink will communicate the uuid and bucket name with the tardigrade satellite, checking for a previous attribution. If no attribution is found on the specified bucket and the bucket is currently void of data, the satellite will attribute the partners uuid to that bucket within the metadata struct.
+
+After the connector has confirmed the partners uuid was successfully attributed to a bucket, the connector will notify a tardigrade satellite that the attribution was confirmed. The satellite will then look up the bucket, verify that partner has the attribution and add the necessary data to the Attribution table. 
```

Comment:
isn't the previous step already enough? the satellite knows of the connection. what is the separate operation for?

Reply:
Separation of concerns. If we had the satellite update the value attribution database when the metadata is added, we need to include information about value attribution within the meta service.

Reply:
that's fine but the performance hit is too costly. this adds another round trip from the client to the satellite.

Reply:
👍

Reply:
Also if the first RPC call succeeds but the file upload fails we have more potential dangling entries, am I understanding that correctly? Combining it into the existing RPC call makes that less possible?

Reply:
After talking with @mniewrzal I don't believe we will add it to the metadata struct anyways, since it could be changed by the user.